### PR TITLE
[FLINK-2918] Add method to read a file of type SequenceFileInputFormat

### PR DESCRIPTION
**Pull request by smarthi**
Fixes FLINK-2918

### CODE REVIEW

**Comment by tillrohrmann** [comment]
LGTM. The `readAsSequenceFile` is nice syntactic sugar to add. But we should add documentation to the online documentation since it's the user facing API which we extend here.

Furthermore, we should think about adding a corresponding `writeAsSequenceFile` method to the `DataSet` as the counter-part to `readAsSequenceFile`.

**Comment by smarthi** [comment]
Also need to add a method for the new Hadoop API, the present PR only deals with Hadoop v1

**Comment by smarthi** [comment]
@tillrohrmann There's an existing DataSet#write() api, don't think we need a DataSet#writeAsSequenceFile().

**Comment by fhueske** [comment]
Hi @smarthi,
I am not super familiar with Hadoop interface, but is the `hadoop2` switch really necessary? Hadoop2 does also support the `mapred.*` interfaces, right? Or are both `SequenceInputFormat`s and `SequenceOutputFormat`s not compatible?

Thanks, Fabian

**Comment by smarthi** [comment]
@fhueske Updated ProgrammingGuide.md to reflect the missing APIs

File: flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java [hunk 1]
```diff
@@ -580,6 +580,14 @@ public CsvReader readCsvFile(String filePath) {
 	}
 
 	/**
+	 * Creates a {@link DataSet} from {@link org.apache.hadoop.mapred.SequenceFileInputFormat}
+	 * A {@link org.apache.hadoop.mapred.JobConf} with the given inputPath is created.
+ 	 */
+	public <K,V> DataSource<Tuple2<K, V>> readAsSequenceFile(Class<K> key, Class<V> value, String inputPath) throws IOException {
```
Comment (fhueske) [inline flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java]: Can we rename the method to `readSequenceFile()` to have it consistent with the other methods (`readCsvFile`, `readHadoopFile`, ...)

File: flink-scala/src/main/scala/org/apache/flink/api/scala/ExecutionEnvironment.scala [hunk 2]
```diff
@@ -440,6 +438,19 @@ class ExecutionEnvironment(javaEnv: JavaEnv) {
   }
 
   /**
+   * Creates a [[DataSet]] from [[org.apache.hadoop.mapred.SequenceFileInputFormat]]
+   * A [[org.apache.hadoop.mapred.JobConf]] with the given inputPath is created.
+   */
+  def readAsSequenceFile[K, V](
```
Comment (fhueske) [inline flink-scala/src/main/scala/org/apache/flink/api/scala/ExecutionEnvironment.scala]: Rename to `readSequenceFile`?

File: docs/apis/programming_guide.md [hunk 3]
```diff
@@ -1726,6 +1726,13 @@ File-based:
    
 - `readFileOfPrimitives(path, delimiter, Class)` / `PrimitiveInputFormat` - Parses files of new-line (or another char sequence)
    delimited primitive data types such as `String` or `Integer` using the given delimiter.
+   
+- `readHadoopFile(FileInputFormat, Key, Value, path)` / `FileInputFormat` - Creates a JobConf and reads file from the spacified
```
Comment (fhueske) [inline docs/apis/programming_guide.md]: spacified -> specified

File: docs/apis/programming_guide.md [hunk 4]
```diff
@@ -1726,6 +1726,13 @@ File-based:
    
 - `readFileOfPrimitives(path, delimiter, Class)` / `PrimitiveInputFormat` - Parses files of new-line (or another char sequence)
    delimited primitive data types such as `String` or `Integer` using the given delimiter.
+   
+- `readHadoopFile(FileInputFormat, Key, Value, path)` / `FileInputFormat` - Creates a JobConf and reads file from the spacified 
+   path with the specified FileInputFormat, Key class and Value Class and returns them as Tuple2<Key, Value>.
```
Comment (fhueske) [inline docs/apis/programming_guide.md]: Value Class -> Value class

**Comment by fhueske** [comment]
Thanks for the fast update :-)
You only updated the Java part of the documentation. The Scala part follows the Java part around line 1858 in programming_guide.md. Can copy your Java modifications and port the code examples to Scala?

Thanks a lot, Fabian

File: docs/apis/programming_guide.md [hunk 5]
```diff
@@ -1772,9 +1779,16 @@ DataSet<Tuple2<String, Double>> csvInput = env.readCsvFile("hdfs:///the/CSV/file
 
 // read a CSV file with three fields into a POJO (Person.class) with corresponding fields
 DataSet<Person>> csvInput = env.readCsvFile("hdfs:///the/CSV/file")
-                         .pojoType(Person.class, "name", "age", "zipcode");                         
+                         .pojoType(Person.class, "name", "age", "zipcode");  
+                                                 
 
-// create a set from some given elements
+// read a file from the specified path of type TextInputFormat 
+DataSet<Tuple2<Text, Text>> tuples = env.readHadoopFile(new TextInputFormat(), Text.class, Text.class, "hdfs://nnHost:nnPort/path/to/file");
```
Comment (fhueske) [inline docs/apis/programming_guide.md]: I think TextInputFormat produces (LongWritable, Text)

File: docs/apis/programming_guide.md [hunk 6]
```diff
@@ -1923,6 +1943,12 @@ val values = env.fromElements("Foo", "bar", "foobar", "fubar")
 val numbers = env.generateSequence(1, 10000000);
 {% endhighlight %}
 
+// read a file from the specified path of type TextInputFormat 
+val tuples = env.readHadoopFile(new TextInputFormat, classOf[Text], classOf[Text], "hdfs://nnHost:nnPort/path/to/file")
```
Comment (fhueske) [inline docs/apis/programming_guide.md]: first Text -> LongWritable

**Comment by fhueske** [comment]
Thanks for the quick updates @smarthi!
Looks good to merge, IMO.

File: flink-scala/src/main/scala/org/apache/flink/api/scala/hadoop/mapred/HadoopInputFormat.scala [hunk 7]
```diff
@@ -27,6 +27,10 @@ class HadoopInputFormat[K, V](
     job: JobConf)
   extends HadoopInputFormatBase[K, V, (K, V)](mapredInputFormat, keyClass, valueClass, job) {
 
+  def this(mapredInputFormat: InputFormat[K, V], keyClass: Class[K],valueClass: Class[V]) = {
```
Comment (chiwanpark) [inline flink-scala/src/main/scala/org/apache/flink/api/scala/hadoop/mapred/HadoopInputFormat.scala]: Please add a space after comma.

**Comment by chiwanpark** [comment]
LGTM except one minor cosmetic issue. :+1:

**Comment by smarthi** [comment]
@chiwanpark Fixed it

**Comment by fhueske** [comment]
Will merge this PR

**Pull request closed without merging**